patient_data_source.py

```diff
@@ -0,0 +1,214 @@
+from abc import ABC, abstractmethod
+import names
+import warnings
+from datetime import datetime
+from enum import Enum
+from collections.abc import Iterable
+from fhirclient.models.patient import Patient as FHIR_Patient
+from fhirclient.models.observation import Observation as FHIR_Observation
+from fhirclient.models.fhirdate import FHIRDate
+
+class Patient(ABC):
+	class Gender(Enum):
+		male = 0
+		female = 1
+		other = 2
+		unknown = 3
+
+	def get_gender(self) -> Gender:
+		"""Returns the patient's gender. Default implementation return Gender.unkown"""		
+		return Gender.unknown
+
+	def get_identifier_value(self) -> str:
+		"""Returns a id value for external use."""
+		return None
+
+	@abstractmethod
+	def get_dob(self) -> datetime:
+		"""Returns the patient's date of birth."""
+		return datetime.min
+
+	def get_identifier_system(self) -> str:
+		"""Returns the namespace for unique identifier values."""
+		return None
+
+	def generate_name(self, gender : Gender) -> tuple[str, str]:
+		"""Generate a first and last name based on gender."""
```

Comment:
Make clear in this docstring that the return is last name and THEN first name, in that order.

[See some style guidelines and example docstrings here.](https://google.github.io/styleguide/pyguide.html#doc-function-returns)